llvm/lib/CodeGen/MachineInstr.cpp

```diff
@@ -1478,6 +1479,15 @@ bool MachineInstr::mayAlias(AAResults *AA, const MachineInstr &Other,
   return false;
 }
 
+bool MachineInstr::mayAlias(AAResults *AA, const MachineInstr &Other,
```

Comment:
Where is the non-BatchAA overload used now?

Reply:
`MachineSink.cpp` and several target-specific passes. They probably can also be updated to the batch overload , but this seemed like a reasonable cut off for the first patch. I wonder if this overload should be tagged with `LLVM_DEPRECATED`?

Reply:
Okay, that's fine. And no, you can't tag something as deprecated before all in-tree uses have been removed (it will break buildbots compiling with -Werror).